src/vivarium/interface/interactive.py

```diff
@@ -1,45 +1,99 @@
+import warnings
 from math import ceil
+from typing import Mapping, List, Callable
+
+import pandas as pd
 
 from vivarium.framework.configuration import build_simulation_configuration, build_model_specification
 from vivarium.framework.plugins import PluginManager
+from vivarium.framework.time import Timedelta, Time
+from vivarium.framework.values import Pipeline
 from vivarium.framework.engine import SimulationContext
-from vivarium.config_tree import ConfigTree
 
 from .utilities import run_from_ipython, log_progress, raise_if_not_setup
 
-from typing import Mapping, List
-
 
 class InteractiveContext(SimulationContext):
+    """A simulation context with helper methods for running simulations interactively.
+
+    This class should not be instantiated directly. It should be created with a call to one
+    of the  helper methods provided in this module.
+    """
 
     def __init__(self, configuration, components, plugin_manager=None):
         super().__init__(configuration, components, plugin_manager)
         self._initial_population = None
 
     def setup(self):
+        """Setup the simulation and initialize its population."""
         super().setup()
         self._start_time = self.clock.time
         self.initialize_simulants()
 
     def initialize_simulants(self):
+        """Initialize this simulation's population. Should not be called directly."""
         super().initialize_simulants()
         self._initial_population = self.population.get_population(True)
 
     def reset(self):
-        # This is super crude, but should work for a great deal of components.
+        """Reset the simulation to its initial state."""
+        warnings.warn("This reset method is very crude.  It should work for "
+                      "many simple simulations, but we make no guarantees. In "
+                      "particular, if you have components that manage their "
+                      "own state in any way, this might not work.")
         self.population._population = self._initial_population
         self.clock._time = self._start_time
 
     @raise_if_not_setup(system_type='run')
-    def run(self, with_logging=True):
+    def run(self, with_logging: bool=True) -> int:
```

Comment:
should we also type hint None return types or do we not care as much about that?

Reply:
I vote for no